sign/rpmgensig.c

```diff
@@ -276,7 +296,17 @@ static int runGPG(sigTarget sigt, const char *sigfile)
     rpmPopMacro(NULL, "__plaintext_filename");
     rpmPopMacro(NULL, "__signature_filename");
 
-    fnamedPipe = Fopen(namedPipeName, "w");
+    while (1) {
```

Comment:
Looping while fnamedPipe remains NULL might be a more natural check here, eternal loops always make me a bit fidgety.